src/main/java/org/broadinstitute/hellbender/tools/spark/CreateIndexForHadoopBam.java

```diff
@@ -0,0 +1,158 @@
+            optional = false )
+    public File inputBam;
+
+    @Argument(fullName = StandardArgumentDefinitions.READ_VALIDATION_STRINGENCY_LONG_NAME,
+            shortName = StandardArgumentDefinitions.READ_VALIDATION_STRINGENCY_SHORT_NAME,
+            doc = "Validation stringency for all SAM/BAM/CRAM/SRA files read by this program.  The default stringency value SILENT " +
+                    "can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) " +
+                    "do not otherwise need to be decoded.",
+            common=true,
+            optional=true)
+    public ValidationStringency readValidationStringency = ReadConstants.DEFAULT_READ_VALIDATION_STRINGENCY;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc = "The BAM splitting_index file. If this is unspecified an index will be created with the same name as " +
+                    "the input file but with the additional extension " + SplittingBAMIndexer.OUTPUT_FILE_EXTENSION,
+            optional = true)
+    public File output;
+
+    @Argument(fullName = SPLITTING_INDEX_GRANULARITY_LONG_NAME,
+            shortName = SPLITTING_INDEX_GRANULARITY_SHORT_NAME,
+            doc = "Splitting index granularity, an entry is created in the index every this many reads.",
+            optional = true)
+    public int granularity = SplittingBAMIndexer.DEFAULT_GRANULARITY;
+
+    @Argument(fullName = CREATE_BAI_LONG_NAME,
+            shortName = CREATE_BAI_SHORT_NAME,
+            doc = "set this to create a bai index at the same time as creating a splitting index",
+            optional = true)
+    public boolean createBai = false;
+
+
+    @Override
+    protected Object doWork() {
+        if( granularity <= 0) {
+            throw new UserException.BadArgumentValue(SPLITTING_INDEX_GRANULARITY_LONG_NAME, Integer.toString(granularity), "Granularity must be > 0");
+        }
+        final File index = getOutputFile(output, inputBam);
+        if(createBai){
+            createBaiAndSplittingIndex(inputBam, index, granularity, readValidationStringency);
+        } else {
+            createOnlySplittingIndex(inputBam, index, granularity);
+        }
+
+        return 0;
+    }
+
+    private static void createOnlySplittingIndex(File inputBam, File index, int granularity) {
+        assertIsBam(inputBam);
+        //createBamSplittingIndex(inputBam, getOutputFile(output, inputBam), readValidationStringency, granularity);
+        try(BufferedInputStream in = new BufferedInputStream(new FileInputStream(inputBam))){
+            try(BufferedOutputStream out = new BufferedOutputStream(new FileOutputStream(index))) {
+                SplittingBAMIndexer.index(in, out, inputBam.length(), granularity);
+            } catch (IOException e){
+                throw new UserException.CouldNotCreateOutputFile(index, e);
```

Comment:
this is true that this exception is only related to creating the open file , ie is this accurate description of what happened (CouldNotCreateOutputFile) - I dont think it is. The underlying IOException may be caused by a whole bunch of stuff